### ISSUE TITLE
No Questions around

### ISSUE THREAD

**Issue by Mannshoch** [westnordost/StreetComplete#1542]
After the update to v13 a lot of ne quests appear in the city i life in.
Now I did a litle bicycle tour out of the city around 8-10km direct distance - 15km  tour through around 4 villages .
I know from OSM there  are really much open quests. But I got none. We drove around an hour but i got only  three quests to answer and another three not on our way. "Search here" not worked. The red bar stoped in the middle for minutes and nothing happend.

**Versions affected**
13

**Comment by matkoniecz** [comment]
> The red bar stoped in the middle for minutes and nothing happend.

So SC is trying to download quests but it is for some reason going really slow. Is your internet working? Can you maybe share location where this happened? Maybe something weird happened there with OSM data.

**Comment by Mannshoch** [comment]
08-27 12:05:19.946 I/ActivityManager(531): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=de.westnordost.streetcomplete/.MainActivity bnds=[540,1389][792,1685]} from uid 10170
08-27 12:05:20.018 I/ActivityManager(531): Start proc 13674:de.westnordost.streetcomplete/u0a90 for activity de.westnordost.streetcomplete/.MainActivity
08-27 12:05:20.610 W/SurfaceFlinger(405): Attempting to set client state on removed layer: Surface(name=AppWindowToken{ca6f4f9 token=Token{327dfc0 ActivityRecord{7582d43 u0 de.westnordost.streetcomplete/.MainActivity t207}}})/@0x1829884 - animation-leash#0
08-27 12:05:20.611 W/SurfaceFlinger(405): Attempting to destroy on removed layer: Surface(name=AppWindowToken{ca6f4f9 token=Token{327dfc0 ActivityRecord{7582d43 u0 de.westnordost.streetcomplete/.MainActivity t207}}})/@0x1829884 - animation-leash#0
08-27 12:05:20.697 W/System.err(13674): 	at de.westnordost.streetcomplete.tangram.MapFragment.getMapAsync(MapFragment.java:146)
08-27 12:05:20.697 W/System.err(13674): 	at de.westnordost.streetcomplete.tangram.QuestsMapFragment.getMapAsync(QuestsMapFragment.java:135)
08-27 12:05:20.697 W/System.err(13674): 	at de.westnordost.streetcomplete.tangram.MapFragment.getMapAsync(MapFragment.java:128)
08-27 12:05:20.697 W/System.err(13674): 	at de.westnordost.streetcomplete.MainActivity.onCreate(MainActivity.java:255)
08-27 12:05:21.456 W/.streetcomplet(13674): Long monitor contention with owner Thread-3 (13698) at int android.graphics.drawable.VectorDrawable.nDraw(long, long, long, android.graphics.Rect, boolean, boolean)(VectorDrawable.java:-2) waiters=0 in java.util.List de.westnordost.streetcomplete.tangram.TangramQuestSpriteSheetCreator.get() for 737ms
08-27 12:05:21.572 D/Tangram (13674): WARNING sceneLoader.cpp:1745: Can't find data source streetcomplete_quests for layer streetcomplete_quests
08-27 12:05:21.572 D/Tangram (13674): WARNING sceneLoader.cpp:1745: Can't find data source streetcomplete_geometry for layer streetcomplete_geometry
08-27 12:05:21.645 I/ActivityManager(531): Start proc 13730:de.westnordost.streetcomplete:lost/u0a90 for service de.westnordost.streetcomplete/com.mapzen.android.lost.internal.FusedLocationProviderService
08-27 12:05:22.283 W/SurfaceFlinger(405): Attempting to set client state on removed layer: Splash Screen de.westnordost.streetcomplete#0
08-27 12:05:22.283 W/SurfaceFlinger(405): Attempting to destroy on removed layer: Splash Screen de.westnordost.streetcomplete#0
08-27 12:05:24.308 W/NotificationService(531): Toast already killed. pkg=de.westnordost.streetcomplete callback=android.app.ITransientNotification$Stub$Proxy@be2d5b0
08-27 12:05:27.236 W/NotificationService(531): Toast already killed. pkg=de.westnordost.streetcomplete callback=android.app.ITransientNotification$Stub$Proxy@def0ae0
08-27 12:05:58.730 W/NotificationService(531): Toast already killed. pkg=de.westnordost.streetcomplete callback=android.app.ITransientNotification$Stub$Proxy@2dd910f
08-27 12:06:00.719 W/NotificationService(531): Toast already killed. pkg=de.westnordost.streetcomplete callback=android.app.ITransientNotification$Stub$Proxy@29a97a0
08-27 12:06:29.184 W/NotificationService(531): Toast already killed. pkg=de.westnordost.streetcomplete callback=android.app.ITransientNotification$Stub$Proxy@e4987a9
08-27 12:06:34.130 W/NotificationService(531): Toast already killed. pkg=de.westnordost.streetcomplete callback=android.app.ITransientNotification$Stub$Proxy@f0adf78
08-27 12:06:40.266 W/SurfaceFlinger(405): Attempting to set client state on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:06:40.266 W/SurfaceFlinger(405): Attempting to destroy on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:06:40.270 W/SurfaceFlinger(405): Attempting to set client state on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:06:40.270 W/SurfaceFlinger(405): Attempting to destroy on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:08:38.696 I/ActivityManager(531): START u0 {cmp=de.westnordost.streetcomplete/.MainActivity} from uid 10090
08-27 12:08:38.697 W/ActivityManager(531): startActivity called from non-Activity context; forcing Intent.FLAG_ACTIVITY_NEW_TASK for: Intent { cmp=de.westnordost.streetcomplete/.MainActivity }
08-27 12:08:38.715 W/WindowManager(531): TaskSnapshotSurface.create: Failed to find main window for token=AppWindowToken{9b5396f token=Token{fe7344e ActivityRecord{d97bf13 u0 de.westnordost.streetcomplete/.MainActivity t207}}}
08-27 12:08:38.791 W/System.err(13674): 	at de.westnordost.streetcomplete.tangram.MapFragment.getMapAsync(MapFragment.java:146)
08-27 12:08:38.791 W/System.err(13674): 	at de.westnordost.streetcomplete.tangram.QuestsMapFragment.getMapAsync(QuestsMapFragment.java:135)
08-27 12:08:38.791 W/System.err(13674): 	at de.westnordost.streetcomplete.tangram.MapFragment.getMapAsync(MapFragment.java:128)
08-27 12:08:38.791 W/System.err(13674): 	at de.westnordost.streetcomplete.MainActivity.onCreate(MainActivity.java:255)
08-27 12:08:38.951 D/Tangram (13674): WARNING sceneLoader.cpp:1745: Can't find data source streetcomplete_quests for layer streetcomplete_quests
08-27 12:08:38.951 D/Tangram (13674): WARNING sceneLoader.cpp:1745: Can't find data source streetcomplete_geometry for layer streetcomplete_geometry
08-27 12:08:39.503 I/ActivityManager(531): Displayed de.westnordost.streetcomplete/.MainActivity: +666ms
08-27 12:08:39.952 W/SurfaceFlinger(405): Attempting to set client state on removed layer: Surface(name=AppWindowToken{9b5396f token=Token{fe7344e ActivityRecord{d97bf13 u0 de.westnordost.streetcomplete/.MainActivity t207}}})/@0x33c7c55 - animation-leash#0
08-27 12:08:39.952 W/SurfaceFlinger(405): Attempting to destroy on removed layer: Surface(name=AppWindowToken{9b5396f token=Token{fe7344e ActivityRecord{d97bf13 u0 de.westnordost.streetcomplete/.MainActivity t207}}})/@0x33c7c55 - animation-leash#0
08-27 12:08:41.166 W/NotificationService(531): Toast already killed. pkg=de.westnordost.streetcomplete callback=android.app.ITransientNotification$Stub$Proxy@b9c01ca
08-27 12:08:44.336 W/SurfaceFlinger(405): Attempting to set client state on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:08:44.336 W/SurfaceFlinger(405): Attempting to destroy on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:08:44.342 W/SurfaceFlinger(405): Attempting to set client state on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:08:44.342 W/SurfaceFlinger(405): Attempting to destroy on removed layer: de.westnordost.streetcomplete/de.westnordost.streetcomplete.MainActivity#0
08-27 12:08:49.229 I/ActivityManager(531): Process de.westnordost.streetcomplete:lost (pid 13730) has died: cch+4CEM 
08-27 12:09:45.725 E/QuestDownload(13674): 	at de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao.get(OverpassMapDataDao.java:60)
08-27 12:09:45.725 E/QuestDownload(13674): 	at de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao.getAndHandleQuota(OverpassMapDataDao.java:84)
08-27 12:09:45.725 E/QuestDownload(13674): 	at de.westnordost.streetcomplete.quests.place_name.AddPlaceName.download(AddPlaceName.kt:65)
08-27 12:09:45.725 E/QuestDownload(13674): 	at de.westnordost.streetcomplete.data.osm.download.OsmQuestDownload.download(OsmQuestDownload.java:90)
08-27 12:09:45.725 E/QuestDownload(13674): 	at de.westnordost.streetcomplete.data.download.QuestDownload.downloadQuestTypes(QuestDownload.java:200)
08-27 12:09:45.725 E/QuestDownload(13674): 	at de.westnordost.streetcomplete.data.download.QuestDownload.download(QuestDownload.java:124)
08-27 12:09:45.725 E/QuestDownload(13674): 	at de.westnordost.streetcomplete.data.download.QuestDownloadService$ServiceHandler.handleMessage(QuestDownloadService.java:207)

**Comment by MegaArthur** [comment]
> 08-27 12:09:45.725 E/QuestDownload(13674): at de.westnordost.streetcomplete.quests.place_name.AddPlaceName.download(AddPlaceName.kt:65)

Seems like this error might be related to AddPlaceName being slow. I think once osm-performance is merged into master and a new release is done, it should work fine.

For now you can try disabling the AddPlaceName quest (and other quests that take a long time, as mentioned in #1514)

**Comment by Mannshoch** [comment]
The first test was great. Now "search here" works. I deactivated "Add Place" name. I will test it further.

**Comment by westnordost** [comment]
Add Place Name was among those that were extremely slow, see #1514. No need to investigate this further until v14.0 is out. You can post here again if it should occur with v14.0 again too.
